packages/sitecore-jss/src/site/graphql-sitemap-service.ts

```diff
@@ -0,0 +1,103 @@
+import { GraphQLClient, GraphQLRequestClient } from '../graphql';
+import { siteNameError } from '../constants';
+import debug from '../debug';
+
+const PREFIX_NAME_SITEMAP = 'sitemap';
+
+// The default query for request sitemaps
+const defaultQuery = /* GraphQL */ `
+  query SitemapQuery($siteName: String!) {
+    site {
+      siteInfo(site: $siteName) {
+        sitemap
+      }
+    }
+  }
+`;
+
+export type GraphQLSitemapServiceConfig = {
+  /**
+   * Your Graphql endpoint
+   */
+  endpoint: string;
+  /**
+   * The API key to use for authentication
+   */
+  apiKey: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
+};
+
+/**
+ * The schema of data returned in response to sitemaps request
+ */
+export type SitemapQueryResult = { site: { siteInfo: { sitemap: string[] } } };
+
+/**
+ * Service that fetch the sitemaps data using Sitecore's GraphQL API.
+ */
+export class GraphQLSitemapService {
+  private graphQLClient: GraphQLClient;
+
+  protected get query(): string {
+    return defaultQuery;
+  }
+
+  /**
+   * Creates an instance of graphQL sitemaps service with the provided options
+   * @param {GraphQLSitemapServiceConfig} options instance
+   */
+  constructor(public options: GraphQLSitemapServiceConfig) {
+    this.graphQLClient = this.getGraphQLClient();
+  }
+
+  /**
+   * Fetch a data of sitemap from API
+   * @returns text of sitemap
+   * @throws {Error} if the siteName is empty.
+   */
+  async fetchSitemap(): Promise<string[]> {
+    const siteName: string = this.options.siteName;
+
+    if (!siteName) {
+      throw new Error(siteNameError);
+    }
+
+    const sitemapResult: Promise<SitemapQueryResult> = this.graphQLClient.request(this.query, {
+      siteName,
+    });
+    try {
+      return sitemapResult.then((result: SitemapQueryResult) => result.site.siteInfo.sitemap);
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+
+  /**
+   * return exists sitemap
+   * @param idSitemap{string}
+   * @returns Promise{string | undefined}
```

Comment:
Could use better jsdoc comments here. Something like
```suggestion
   * Get sitemap file path for sitemap id
   * @param {string} id the sitemap id (can be empty for default 'sitemap.xml' file)
   * @returns {string | undefined} the sitemap file path or undefined if one doesn't exist
```